### PR TITLE
Split container parents should be redrawn when swapping child containers (within parent)

### LINKED ISSUE #4764: Parent split container titles not updating on child con swap/move for some parent (of parent) con layouts
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Parent split con titles do not update correctly when moving child (sibling) container in some parent layouts.

## Expected Behavior

Parent split con titles (and parents of parent split con titles) should always update / reflect current children tree representations.

## Reproduction Instructions

1. Create a `H[H[app1 app2]]` split con layout with two different applications, e.g.:
image
I've found this layout that exhibits this behaviour, or seemingly anything with a parent horizontal layout with child tabbed, stacked, vert, etc. layout also exhibit this behaviour.
2. select one of the applications and move it left or right (or up/down) within the split container (parent);
3. look at the parent container title while moving said applications (they do not update for this parent/child layout).

See below for examples:

one on i3 version `4.20.1-2-gd83940a8` and the next with a fix applied (see [PR 4765](https://github.com/i3/i3/pull/4765)).
### current: (look at the split con title which is not updating)
split-parent-not-updated

### with proposed fix applied (see this [PR 4765](https://github.com/i3/i3/pull/4765))
split-parent-updated-fixed

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.20.1-2-gd83940a8 © 2009 Michael Stapelberg and contributors
Running i3 version: 4.20.1-2-gd83940a8 (pid 1742369)
Loaded i3 config:
  /home/jay/.i3/config (main) (last modified: Sat 01 Jan 2022 07:30:47, 30526 seconds ago)

The i3 binary you just called: /usr/local/bin/i3
The i3 binary you are running: i3-with-al/bin/i3
</pre>

<details><summary>Config file</summary><pre>
</pre>
</details>

<pre>
Logfile URL: https://logs.i3wm.org/logs/6208202748198912.bz2
</pre>

<pre>
- Linux Distribution & Version: Manjaro 21.2.0 Qonos
- Are you using a compositor (e.g., xcompmgr or compton): picom
</pre>

### PR DESCRIPTION
**Pull request by jtaala**
This is a proposed simple fix for submitted issue [4764](https://github.com/i3/i3/issues/4764).

* adds a `con_force_split_parents_redraw(con)` function call to the tree_move function when child move is within parent container.
* fixes #4764

### CODE REVIEW

**Comment by jtaala** [comment]
simplified the in-code comment (was way too wordy and not quite accurate).

**Comment by orestisfl** [comment]
LGTM but please include a line for the changelog: https://github.com/i3/i3/blob/next/.github/CONTRIBUTING.md

**Comment by jtaala** [comment]
> LGTM but please include a line for the changelog: https://github.com/i3/i3/blob/next/.github/CONTRIBUTING.md

Thanks @orestisfl! have added changelog (bugfix) entry.

Cheers,

Jay.

**Comment by jtaala** [comment]
@orestisfl, thanks for running a build & test - very strange that it failed.

I've checked out that branch and can successfully dist-build on my local machine (with `meson .. -Ddocs=true -Dmans=true && ninja -v dist`).

As far as I can tell that failure doesn't look like it's from the PR - has anything changed in the `i3wm/travis-base:2022-04-30650643` image? (e.g. does the dist-build run successfully with that image in current next branch?).

Let me know if there's anything I can do to help test/debug the current failure.

Jay.

**Comment by jtaala** [comment]
Morning @orestisfl!

Could you please approve the workflow for test this again?  I've just merged in latest i3/next branch.

All tests pass on my local build (including dist-build).

Regards!

**Comment by jtaala** [comment]
Thanks @orestisfl , this should be able to be merged now.

Let me know if there's anything else that's needed.

**Review by orestisfl — APPROVED** [review]
Thanks!
